### PR TITLE
RestoreCommand nearest framework support

### LINKED ISSUE #611: Newtonsoft.Json 7.0.1-beta3 does not show up as a lock file target
1. install newtonsoft.json 7.0.1-beta3
2. build

newtonsoft.json.dll does not show up in the lock file under targets.

### PR DESCRIPTION
**Pull request by emgarten**
Fixes: https://github.com/NuGet/Home/issues/611

Adding CompareTest to ContentPropertyDefinition to allow finding the nearest framework in scenarios where multiple frameworks are compatible with a project framework, and neither can be ruled out by checking compatibility between the framework options.

### CODE REVIEW

**Comment by emgarten** [comment]
//cc @anurse

File: src/NuGet.ContentModel/ContentPropertyDefinition.cs [hunk 1]
```diff
@@ -146,6 +157,11 @@ public virtual bool TryLookup(string name, out object value)
 
         public Func<object, object, bool> CompatibilityTest { get; }
 
+        /// <summary>
+        /// Find the nearest compatible candidate.
+        /// </summary>
+        public Func<object, object, object, int> CompareTest { get; }
```
Comment (yishaigalatzer) [inline src/NuGet.ContentModel/ContentPropertyDefinition.cs:163]: this looks like a bug farm waiting to happen, and so do all the method calls with a million nulls. Can we track cleanup for U1

**Comment by yishaigalatzer** [comment]
:shipit:

**Comment by analogrelay** [comment]
Looks good to me, :shipit:. @davidfowl , might be good for you to take a quick look too.

File: src/NuGet.ContentModel/ContentPropertyDefinition.cs (hunk 1 above)
Comment (emgarten) [inline src/NuGet.ContentModel/ContentPropertyDefinition.cs:163]: https://github.com/NuGet/Home/issues/684

**Pull request merged**